static/js/components/subject_page/disaster_event_block.tsx

```diff
@@ -108,55 +134,79 @@ export const DisasterEventBlock = memo(function DisasterEventBlock(
       props.eventTypeSpec,
       props.columns
     );
+    const spinnerId = getSpinnerId();
+    if (!_.isNull(disasterEventData)) {
+      loadSpinner(spinnerId);
+    }
     fetchDisasterEventData(
-      props.id,
       blockEventTypeSpecs.current,
       props.place.dcid,
+      date,
+      severityFilters,
       fetchData
     ).then((disasterData) => {
-      setDisasterEventData(disasterData);
+      setDisasterEventData({
+        disasterEventData: disasterData,
+        date,
+        severityFilters,
+      });
+      removeSpinner(spinnerId);
     });
+  }, [props, date, severityFilters]);
 
-    function handleHashChange() {
-      const spinnerId = getSpinnerId();
-      loadSpinner(spinnerId);
-      fetchDisasterEventData(
-        props.id,
-        blockEventTypeSpecs.current,
-        props.place.dcid,
-        fetchData
-      ).then((disasterData) => {
-        setDisasterEventData(disasterData);
-        removeSpinner(spinnerId);
-      });
+  useEffect(() => {
+    // When date or severity filters are updated, update the hash. Unless this
+    // is nl interface or the initial load.
+    if (isInitialLoading.current || isNlInterface()) {
+      isInitialLoading.current = false;
+      return;
     }
-    window.addEventListener("hashchange", handleHashChange);
-    return () => {
-      window.removeEventListener("hashchange", handleHashChange);
-    };
-  }, [props]);
+    const updatedParams = {};
+    if (!_.isEqual(prevFilters.current, severityFilters)) {
+      updatedParams[URL_HASH_PARAM_KEYS.SEVERITY_FILTER] =
+        JSON.stringify(severityFilters);
+      prevFilters.current = severityFilters;
+    }
+    if (!_.isEqual(prevDate.current, date)) {
+      updatedParams[URL_HASH_PARAM_KEYS.DATE] = date;
+      prevDate.current = date;
+    }
+    const updatedHash = getUpdatedHash(updatedParams, props.id);
+    const currentHash = location.hash.replace("#", "");
+    const urlRoot = window.location.href.split("#")[0];
+    if (updatedHash && updatedHash !== currentHash) {
+      history.pushState({}, "", `${urlRoot}#${updatedHash}`);
```

Comment:
if there are multiple blocks in a page, would they result in multiple history state?

Reply:
multiple blocks will all push to the same history state

Reply:
Why is this not at the app level? I would imagine component should not make page level state change?

Reply:
this only updates the hash with the current block's changes .. and the page doesn't know the information about the block level controls so it'll also be tricky to do the update at the page level. Also wondering if we even need to update the hash because that's just so we can copy + paste the link, the app doesn't actually read state from the hash

Reply:
This is not blocking this PR but maybe we can talk more about this offline.